client/blocks/taxonomy-manager/list.jsx

```diff
@@ -0,0 +1,341 @@
+/**
+ * External dependencies
+ */
+import React, { PropTypes } from 'react';
+import { connect } from 'react-redux';
+import classNames from 'classnames';
+import { localize } from 'i18n-calypso';
+import VirtualScroll from 'react-virtualized/VirtualScroll';
+import {
+	debounce,
+	difference,
+	includes,
+	filter,
+	map,
+	memoize,
+	range,
+	reduce,
+} from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import CompactCard from 'components/card/compact';
+import { decodeEntities } from 'lib/formatting';
+import QueryTerms from 'components/data/query-terms';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import {
+	isRequestingTermsForQueryIgnoringPage,
+	getTermsLastPageForQuery,
+	getTermsForQueryIgnoringPage,
+} from 'state/terms/selectors';
+
+/**
+ * Constants
+ */
+const DEFAULT_TERMS_PER_PAGE = 100;
+const LOAD_OFFSET = 10;
+const ITEM_HEIGHT = 55;
+
+const TaxonomyManagerList = React.createClass( {
```

Comment:
We probably should use ES6 class instead ? Discussion in calypso-framework channel.